ddtrace/contrib/cherrypy/middleware.py

```diff
@@ -0,0 +1,149 @@
+"""
+Datadog trace code for cherrypy.
+"""
+
+# stdlib
+import logging
+
+# project
+from ddtrace import config
+from .. import trace_utils
+from ... import compat
+from ...ext import errors, SpanTypes
+from ...propagation.http import HTTPPropagator
+from ...utils.formats import asbool, get_env
+
+# 3p
+import cherrypy
+from cherrypy.lib.httputil import valid_status
+
+log = logging.getLogger(__name__)
+
+
+# Configure default configuration
+config._add(
+    "cherrypy",
+    dict(
+        distributed_tracing=asbool(get_env("cherrypy", "distributed_tracing", default=True)),
+    ),
+)
+
+SPAN_NAME = "cherrypy.request"
+
+
+class TraceTool(cherrypy.Tool):
+    def __init__(self, app, tracer, service, use_distributed_tracing):
+        self.app = app
+        self._tracer = tracer
+        self.service_name = service
+        self.use_distributed_tracing = use_distributed_tracing
+
+        # CherryPy uses priority to determine which tools act first on each event. The lower the number, the higher
+        # the priority. See: https://docs.cherrypy.org/en/latest/extend.html#tools-ordering
+        cherrypy.Tool.__init__(self, "on_start_resource", self._on_start_resource, priority=95)
+
+    @property
+    def use_distributed_tracing(self):
```

Comment:
nice! I like this approach - much nicer to deal with one source of truth than multiple 😄